ThreeSum.java

```diff
@@ -0,0 +1,103 @@
+
+// Time Complexity : O(n^2logn)
+// Space Complexity : O(1)
+// Did this code successfully run on Leetcode :
+
+
+//******
+// Any problem you faced while coding this : Can you  please explain time complex and space Complex for the below solutions
+
+
+// Your code here along with comments explaining your approach
+
+class Solution {
+    public List<List<Integer>> threeSum(int[] nums) {
+        
+        //Declare List of Integers List
+        List<List<Integer>> result = new ArrayList<>();
+        
+        //return the empty ArrayList
+        if(nums == null || nums.length == 0) return new ArrayList<>();
+        int n = nums.length;
+        
+        //Sort the unsorted Array O(nlogn)
+        Arrays.sort(nums);
+        
+        //loop till 3rd num from last
+        for(int i=0; i < n-2; i++){
+            
+            // if
+            if(i!=0 && nums[i] == nums[i-1]) continue;
+            
+            int low = i+1;
+            int high = n-1;
+            
+            while(low<high){
```

Comment:
When low and high is too far for example, then we can say this loop will take O(N) time.